mne/viz/_3d.py

```diff
@@ -1717,6 +1717,161 @@ def _get_ps_kwargs(initial_time, require='0.6'):
+    """
+    import matplotlib.pyplot as plt
+    import nibabel as nib
+
+    try:
+        from nilearn.plotting import plot_stat_map, plot_glass_brain
+        from nilearn.image import index_img
+    except ImportError:
+        raise ImportError('This function requires nilearn')
+
+    if mode == 'anat':
+        plot_func = plot_stat_map
+    elif mode == 'glass_brain':
+        plot_func = plot_glass_brain
+    else:
+        raise ValueError('Plotting function must be one of anat | glas_brain.'
+                         'Got %s' % mode)
+
+    def _onclick(event, params):
+        """Callback to manage click on the plot."""
+        ax_x, ax_y, ax_z = params['ax_x'], params['ax_y'], params['ax_z']
+        plot_map_callback = params['plot_map_callback']
+
+        if event.inaxes is params['ax_time']:
+            idx = params['stc'].time_as_index(event.xdata)
+            if params['lx'] is None:
+                params['lx'] = params['ax_time'].axvline(
+                    event.xdata, color='g')
+            else:
+                params['lx'].set_xdata(event.xdata)
+
+            cut_coords = (0, 0, 0)
+            if mode == 'anat':
+                cut_coords = (params['ax_y'].lines[0].get_xdata()[0],
+                              params['ax_x'].lines[0].get_xdata()[0],
+                              params['ax_x'].lines[1].get_ydata()[0])
+            params['ax_x'].clear()
+            params['ax_y'].clear()
+            params['ax_z'].clear()
+            params.update({'img_idx': index_img(img, idx)})
+            params.update({'title':
+                           'LCMV (t=%.3f s.)' % params['stc'].times[idx]})
+            plot_map_callback(
+                params['img_idx'], title=params['title'],
+                cut_coords=cut_coords)
+
+        if mode == 'glass_brain':
+            return fig
+
+        if event.inaxes is ax_x:
+            cut_coords = (params['ax_z'].lines[0].get_xdata()[0],
+                          event.xdata, event.ydata)
+        if event.inaxes is ax_y:
+            cut_coords = (event.xdata,
+                          params['ax_x'].lines[0].get_xdata()[0],
+                          event.ydata)
+        if event.inaxes is ax_z:
+            cut_coords = (event.xdata, event.ydata,
+                          params['ax_x'].lines[1].get_ydata()[0])
+
+        if event.inaxes in [ax_x, ax_y, ax_z]:
+            params['ax_x'].clear()
+            params['ax_y'].clear()
+            params['ax_z'].clear()
+            plot_map_callback(
+                params['img_idx'], title=params['title'],
+                cut_coords=cut_coords)
+
+            # XXX: check lines below
+            cut_coords = apply_trans(linalg.inv(img.affine), cut_coords)
+            cut_coords = np.array([int(round(c)) for c in cut_coords])
+
+            # the affine transformation can sometimes lead to corner
+            # cases near the edges?
+            if np.any(cut_coords < 0):
```

Comment:
Maybe a little bit below zero but I doubt it should be `< -1`. In any case a `np.clip` here might be better in case some values are `>= shape[ii]`, too